appsec/src/helper/remote_config/listeners/asm_features_listener.hpp

```diff
@@ -25,14 +29,32 @@ class asm_features_listener : public listener_base {
     [[nodiscard]] std::unordered_map<std::string_view, protocol::capabilities_e>
     get_supported_products() override
     {
-        return {{"ASM_FEATURES", protocol::capabilities_e::ASM_ACTIVATION}};
+        protocol::capabilities_e capabilities = protocol::capabilities_e::NONE;
+
+        if (dynamic_enablement_) {
+            capabilities = protocol::capabilities_e::ASM_ACTIVATION;
+        }
+        if (api_security_enabled_) {
+            capabilities |=
+                protocol::capabilities_e::ASM_API_SECURITY_SAMPLE_RATE;
+        }
+
+        if (capabilities != protocol::capabilities_e::NONE) {
+            return {{asm_features, capabilities}};
+        }
+        return {};
     }
 
     void init() override {}
     void commit() override {}
 
 protected:
+    static constexpr std::string_view asm_features = "ASM_FEATURES";
+    void parse_asm(const rapidjson::Document &serialized_doc);
+    void parse_api_security(const rapidjson::Document &serialized_doc);
     std::shared_ptr<service_config> service_config_;
```

Comment:
I may be missing something. Service config is used in different places of the system. For example, here its values are set while on the client for example their values are read. It's because of that shared core responsability that it is shared